src/charm.py

```diff
@@ -532,11 +532,14 @@ def _on_bessd_pebble_ready(self, event: EventBase) -> None:
         if not self._kubernetes_multus.is_ready():
             self.unit.status = WaitingStatus("Waiting for Multus to be ready")
             return
-        if not self._bessd_container.exists(path=BESSD_CONTAINER_CONFIG_PATH):
-            self.unit.status = WaitingStatus("Waiting for storage to be attached")
-            return
-        self._configure_bessd_workload()
-        self._set_unit_status()
+        try:
```

Comment:
Why not run can_connect and return if we can't instead of triggering the error?  Here we are surrounding complex methods (`_configure_bessd_workload`, `self._set_unit_status` with a very specific error type.

Reply:
The `can_connect` method gives the availability of Pebble when called, but does not guarantee that Pebble will be available in subsequent calls.